easybuild/easyconfigs/g/GROMACS/GROMACS-2024.2-foss-2023b-CUDA-12.5.0.eb

```diff
@@ -0,0 +1,88 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+#
+# Copyright:: Copyright 2012-2016 University of Luxembourg / LCSB, Cyprus Institute / CaSToRC,
+#                                 Ghent University / The Francis Crick Institute
+# Authors::
+# * Wiktor Jurkowski <wiktor.jurkowski@gmail.com>
+# * Fotis Georgatos <fotis@cern.ch>
+# * George Tsouloupas <g.tsouloupas@cyi.ac.cy>
+# * Kenneth Hoste <kenneth.hoste@ugent.be>
+# * Adam Huffman <adam.huffman@crick.ac.uk>
+# * Ake Sandgren <ake.sandgren@hpc2n.umu.se>
+# * J. Sassmannshausen <Crick HPC team>
+# * Dugan Witherick <dugan.witherick@warwick.ac.uk>
+# * Christoph Siegert <christoph.siegert@uni-leipzig.de>
+# License::   MIT/GPL
+
+name = 'GROMACS'
+version = '2024.2'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://www.gromacs.org'
+description = """
+GROMACS is a versatile package to perform molecular dynamics, i.e. simulate the
+Newtonian equations of motion for systems with hundreds to millions of
+particles.
+
+This is a GPU enabled build, containing both MPI and threadMPI binaries.
+
+It also contains the gmxapi extension for the single precision MPI build.
+"""
+
+toolchain = {'name': 'foss', 'version': '2023b'}
+toolchainopts = {'openmp': True, 'usempi': True}
+
+source_urls = [
+    'https://ftp.gromacs.org/pub/gromacs/',
+    'ftp://ftp.gromacs.org/pub/gromacs/',
+]
+sources = [SOURCELOWER_TAR_GZ]
+patches = [
+    'GROMACS-2023.1_set_omp_num_threads_env_for_ntomp_tests.patch',
+    'GROMACS-2023.1_fix_tests_for_gmx_thread_mpi.patch',
```

Comment:
I was recently looking into GROMACS 2024.2 as well, and was wondering if this patch is still required. https://gitlab.com/gromacs/gromacs/-/merge_requests/4093 may solve the same or at least similar issue. Looks like @akesandgren added this patch, so maybe he knows?

Reply:
GROMACS-2023.1_fix_tests_for_gmx_thread_mpi.patch should still be relevant as long as it passes patching.
Especially the "Don't drop relevant PYTHONPATH and LD_LIBRARY_PATH settings." part is vital.

I haven't looked at 2024.2 but I assume they haven't fixed those parts yet.

And GROMACS-2023.1_set_omp_num_threads_env_for_ntomp_tests.patch is just an extra precaution to keep OMMP_NUM_THREADS and --ntomp in sync.